packages/cms-data-sync/src/utils/rich-text.ts

```diff
@@ -0,0 +1,43 @@
+/* eslint-disable no-console */
+/* eslint-disable no-param-reassign */
+
+import { Document } from '@contentful/rich-text-types';
+import { parseHtml } from 'contentful-html-rich-text-converter';
+
+export const clearParsedHtmlOutput = (htmlDocument: Document) => ({
+  ...htmlDocument,
+  content: htmlDocument?.content
+    .map((node) =>
+      node.nodeType === 'unordered-list' || node.nodeType === 'ordered-list'
+        ? {
+            ...node,
+            content: node.content.filter(
+              (childNode: { nodeType: string }) =>
+                childNode.nodeType === 'list-item',
+            ),
+          }
+        : node,
+    )
+    // The external lib 'parseHtml' we are using may return
+    // an output with incorrect nodeType, that's why
+    // we need to overwrite its type to any and filter
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    .filter((node: any) => node?.nodeType !== 'text'),
+});
+
+export const convertText = async <T extends { 'en-US': Document | null }>(
+  payload: T,
+  text: string | null,
+  id: string,
+) => {
+  if (text) {
```

Comment:
im of an opinion that such conditionals should be done before sending it to the function, which means your text should not be null ever

Reply:
nice, added in here https://github.com/yldio/asap-hub/pull/2290/commits/796388b7b47a7b5ded531880f0b5404b4f3eec03 + tests fixes here https://github.com/yldio/asap-hub/pull/2290/commits/9daf1b73701bc0f0c9ccaf7412eacb81cbdfc4ce